dashboard/test/mailers/previews/pd_workshop_mailer_preview.rb

```diff
@@ -0,0 +1,70 @@
+class Pd::WorkshopMailerPreview < ActionMailer::Preview
+  include FactoryGirl::Syntax::Methods
+
+  DEFAULT_COURSE = Pd::Workshop::COURSE_ECS
+  DEFAULT_SUBJECT = Pd::Workshop::SUBJECT_ECS_PHASE_2
+
+  # Dynamically define teacher_enrollment_receipt methods with all available partials
+  Pd::WorkshopMailer::DETAILS_PARTIALS.each do |course, partials_by_subject|
+    partials_by_subject.each do |subject, partial|
+      define_method("teacher_enrollment_receipt__#{partial}") do
+        mail :teacher_enrollment_receipt, course, subject
+      end
+    end
+  end
+
+  def teacher_enrollment_receipt__csf
+    mail :teacher_enrollment_receipt, Pd::Workshop::COURSE_CSF
+  end
```

Comment:
Not a convention. It's my attempt to make the names more readable in the list of previews (see screenshot in PR description), with a clear separation between mail name and course/subject.